packages/dds/tree/src/core/tree/detachedFieldIndexCodec.ts

```diff
@@ -4,56 +4,114 @@
  */
 
 import { assert } from "@fluidframework/core-utils";
+import { IIdCompressor } from "@fluidframework/id-compressor";
 import { ICodecOptions, IJsonCodec } from "../../codec";
 import { fail, forEachInNestedMap, setInNestedMap } from "../../util";
+import { EncodedRevisionTag, RevisionTagCodec } from "../rebase";
 import { ForestRootId } from "./detachedFieldIndex";
 import { Format, Versioned, version } from "./detachedFieldIndexFormat";
 import { DetachedFieldSummaryData, Major, Minor } from "./detachedFieldIndexTypes";
 
-export function makeDetachedNodeToFieldCodec({
-	jsonValidator: validator,
-}: ICodecOptions): IJsonCodec<DetachedFieldSummaryData, string> {
-	const versionedValidator = validator.compile(Versioned);
-	const formatValidator = validator.compile(Format);
-	return {
-		encode: (data: DetachedFieldSummaryData): string => {
-			const detachedNodeToFieldData: [Major, Minor, ForestRootId][] = [];
-			forEachInNestedMap(data.data, (root, key1, key2) => {
-				detachedNodeToFieldData.push([key1, key2, root]);
-			});
-			const encoded = {
-				version,
-				data: detachedNodeToFieldData,
-				maxId: data.maxId,
-			};
-			assert(
-				versionedValidator.check(encoded),
-				0x7ff /* Encoded detachedNodeToField data should be versioned */,
-			);
-			assert(formatValidator.check(encoded), 0x800 /* Encoded schema should validate */);
-			return JSON.stringify(encoded);
-		},
-		decode: (data: string): DetachedFieldSummaryData => {
-			const parsed = JSON.parse(data);
-
-			if (!versionedValidator.check(parsed)) {
-				fail("invalid serialized data: did not have a version");
-			}
-			// When more versions exist, we can switch on the version here.
-			if (parsed.version !== version) {
-				fail("Unexpected version for serialized data");
-			}
-			if (!formatValidator.check(parsed)) {
-				fail("Serialized data failed validation");
-			}
-			const map = new Map();
-			for (const [major, minor, root] of parsed.data) {
-				setInNestedMap(map, major, minor, root);
-			}
-			return {
-				data: map,
-				maxId: parsed.maxId,
-			};
-		},
-	};
+class MajorCodec implements IJsonCodec<Major> {
+	private readonly revisionTagCodec: RevisionTagCodec;
+	public constructor(
+		private readonly idCompressor: IIdCompressor,
+		private readonly options: ICodecOptions,
+	) {
+		this.revisionTagCodec = new RevisionTagCodec(idCompressor);
+	}
+
+	public encode(major: Major) {
+		if (major === undefined) {
+			return null;
+		}
+		const id = this.revisionTagCodec.encode(major);
+		/**
+		 * Note that the decode path must provide a session id in which to interpret the revision tag.
```

Comment:
This comment (and assert) only makes sense if this codec is only used for summarization (and never for ops). I assume this is true? If so, it would be good to explicitly state that.

Reply:
yes this is true. added.